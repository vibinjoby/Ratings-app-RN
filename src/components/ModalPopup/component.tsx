import React from 'react'
import { View, Text } from 'react-native'
import Modal from 'react-native-modal'

import Colors from '../../utilities/colors'
import Button from '../Button'
import styles from './styles'

export interface ModalPopupProps {
  testID?: string
  isVisible: boolean
  content: string
  positiveBtnTxt: string
  negativeBtnTxt: string
  onPositiveBtnPress: () => void
  onNegativeBtnPress: () => void
}

const ModalPopup: React.FC<ModalPopupProps> = ({
  testID,
  isVisible,
  content,
  positiveBtnTxt,
  negativeBtnTxt,
  onPositiveBtnPress,
  onNegativeBtnPress,
}: ModalPopupProps) => {
  return (
    <Modal
      testID={testID}
      isVisible={isVisible}
      onBackdropPress={onNegativeBtnPress}
      backdropOpacity={0.5}
      hasBackdrop
    >
      <View style={styles.content} testID="container">
        <Text style={styles.contentTxt}>{content}</Text>
        <View style={styles.btnWrapper} testID="btnWrapper">
          <Button
            title={negativeBtnTxt}
            onPress={onNegativeBtnPress}
            customStyle={styles.cancelBtn}
            textColor={Colors.black}
            testID="negativeBtn"
          />
          <Button
            testID="positiveBtn"
            title={positiveBtnTxt}
            onPress={onPositiveBtnPress}
            customStyle={styles.yesBtn}
          />
        </View>
      </View>
    </Modal>
  )
}

export default ModalPopup
