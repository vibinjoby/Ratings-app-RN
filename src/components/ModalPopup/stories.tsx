import React, { useState } from 'react'
import { View, StyleSheet } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import ModalPopup from './component'
import Button from '../Button'

const Popup = () => {
  const [isModalVisible, setModalVisible] = useState(false)

  const toggleModal = () => {
    setModalVisible((val) => !val)
  }

  return (
    <View style={styles.container}>
      <Button title="Show modal" onPress={toggleModal} />
      <ModalPopup
        isVisible={isModalVisible}
        content="Are you sure you want to delete this user"
        positiveBtnTxt="Yes"
        negativeBtnTxt="No"
        onPositiveBtnPress={() => console.log('positive btn pressed')}
        onNegativeBtnPress={toggleModal}
      />
    </View>
  )
}

storiesOf('Components', module).add('ModalPopup', () => <Popup />)

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})
