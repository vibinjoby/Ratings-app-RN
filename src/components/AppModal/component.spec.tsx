import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import ModalPopup from './component'

describe('ModalPopup', () => {
  const MOCK_FN = jest.fn()

  it('should render', () => {
    const { toJSON } = render(
      <ModalPopup
        isVisible={false}
        content="Are you sure you want to delete this user"
        positiveBtnTxt="Yes"
        negativeBtnTxt="No"
        onPositiveBtnPress={MOCK_FN}
        onNegativeBtnPress={MOCK_FN}
      />,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should be visible', () => {
    const { getByTestId } = render(
      <ModalPopup
        testID="modal"
        isVisible={true}
        content="Are you sure you want to delete this user"
        positiveBtnTxt="Yes"
        negativeBtnTxt="No"
        onPositiveBtnPress={MOCK_FN}
        onNegativeBtnPress={MOCK_FN}
      />,
    )

    expect(getByTestId('modal').props).toHaveProperty('visible', true)
  })

  it('should not be visible', () => {
    const { getByTestId } = render(
      <ModalPopup
        testID="modal"
        isVisible={false}
        content="Are you sure you want to delete this user"
        positiveBtnTxt="Yes"
        negativeBtnTxt="No"
        onPositiveBtnPress={MOCK_FN}
        onNegativeBtnPress={MOCK_FN}
      />,
    )

    expect(getByTestId('modal').props).toHaveProperty('visible', false)
  })
})
