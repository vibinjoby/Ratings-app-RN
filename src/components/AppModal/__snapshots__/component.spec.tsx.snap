// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPopup should render 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    nativeID="animatedComponent"
    onBackdropPress={[MockFunction]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "padding": 15,
        }
      }
      testID="container"
    >
      <Text
        style={
          Object {
            "fontFamily": "Poppins-Regular",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want to delete this user
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginTop": 25,
          }
        }
        testID="btnWrapper"
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E2E2E2",
              "borderRadius": 5,
              "minWidth": 115,
              "opacity": 1,
              "padding": 14,
            }
          }
          testID="negativeBtn"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                },
                Object {
                  "color": "#181725",
                },
              ]
            }
          >
            No
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF5823",
              "borderRadius": 5,
              "minWidth": 115,
              "opacity": 1,
              "padding": 14,
            }
          }
          testID="positiveBtn"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                },
                Object {},
              ]
            }
          >
            Yes
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
